server/protocols/jmap-rfc-8621-integration-tests/distributed-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/distributed/DistributedMailboxSetMethodTest.java

```diff
@@ -64,4 +67,14 @@ public MailboxId randomMailboxId() {
     public String errorInvalidMailboxIdMessage(String value) {
         return String.format("%s is not a mailboxId: Invalid UUID string: %s", value, value);
     }
+
+    @Override
+    @Test
+    @Disabled("Distributed event bus is asynchronous, we cannot expect the newState to be returned immediately after Mailbox/set call")
+    public void newStateShouldBeUpToDate(GuiceJamesServer server) {}
+
+    @Override
+    @Test
+    @Disabled("Distributed event bus is asynchronous, we cannot expect the newState to be returned immediately after Mailbox/set call")
+    public void oldStateShouldIncludeSetChanges(GuiceJamesServer server) {}
```

Comment:
Could not maybe rewrite it, by separating /set and /changes call? You get the old state from the /set response and put an await on your changes call?

Reply:
Thing is at the point when the update completes, the new state might not been stored yet so we can't reliably extract it from the first call. 

If we use await on the state storing process then this test will be a duplication of a similar one from the MailboxChangesMethodContract.